modules/navier_stokes/src/fvkernels/INSFVEnergyTimeDerivative.C

```diff
@@ -19,19 +19,31 @@ INSFVEnergyTimeDerivative::validParams()
   InputParameters params = FVTimeKernel::validParams();
   params.addClassDescription(
       "Adds the time derivative term to the incompressible Navier-Stokes energy equation.");
-  params.addRequiredParam<Real>(NS::density, "The value for the density");
-  params.declareControllable(NS::density);
-  params.addParam<MooseFunctorName>("cp_name", NS::cp, "The name of the specific heat capacity");
+  params.addRequiredParam<MaterialPropertyName>(NS::density, "Density");
+  params.addRequiredParam<MaterialPropertyName>(NS::cp, "Specific heat capacity");
+  params.addParam<MaterialPropertyName>(NS::time_deriv(NS::cp),
+                                        "Specific heat capacity time derivative functor");
   return params;
 }
 
 INSFVEnergyTimeDerivative::INSFVEnergyTimeDerivative(const InputParameters & params)
-  : FVTimeKernel(params), _rho(getParam<Real>(NS::density)), _cp(getFunctor<ADReal>("cp_name"))
+  : FVTimeKernel(params),
+    _rho(getFunctor<ADReal>(getParam<MaterialPropertyName>(NS::density))),
+    _cp(getFunctor<ADReal>(getParam<MaterialPropertyName>(NS::cp))),
+    _cp_dot(isParamValid(NS::time_deriv(NS::cp))
+                ? &getFunctor<ADReal>(getParam<MaterialPropertyName>(NS::time_deriv(NS::cp)))
+                : nullptr)
 {
```

Comment:
add a check that T is an INSFVEnergyVariable ?

Reply:
Added!